src/appleseed-max-impl/appleseedrenderer/tilecallback.cpp

```diff
@@ -264,3 +264,26 @@ void TileCallback::blit_tile(
         }
     }
 }
+
+
+//
+// TileCallbackFactory class implementation.
+//
+
+TileCallbackFactory::TileCallbackFactory(
+    Bitmap*                         bitmap,
+    volatile foundation::uint32*    rendered_tile_count)
+    : m_bitmap(bitmap)
+    , m_rendered_tile_count(rendered_tile_count)
+{
+}
+
+void TileCallbackFactory::release()
+{
+    delete this;
+}
+
+asr::ITileCallback* TileCallbackFactory::create()
+{
+    return new TileCallback(m_bitmap, m_rendered_tile_count);
```

Comment:
That may be incorrect.

There was a reason why we were passing a single tile callback instead of a factory: It's likely that Max doesn't support updating the rendered bitmap or the window from multiple threads, which is exactly what's going to happen with this change.

I don't know if Max supports that. If it doesn't, then that tile callback factory should always return the same tile callback instance (and make sure it only gets destroyed once of course).

Reply:
I think it's better to update bitmap from single thread.
But how returning single instance of TileCallback prevent appleseed from running it from multiple threads?

Reply:
You're of course right, that was stupid of me :) We would need to enqueue the requests and run them from the main thread... Which is exactly what the `SerialTileCallback` does.

It seems to me that maybe the right solution would be to continue using a single tilecallback (instead of a factory), and fix `MasterRenderer` in the sense that it should execute all pending callback updates after adding the render stamp.